src/main/java/ua/softserveinc/tc/controller/admin/DayOffController.java

```diff
@@ -64,6 +68,7 @@
         if (dayOffService.dayOffExist(dayOff.getName(), dayOff.getStartDate())) {
             log.warn("There is another day off with the same name: " + dayOff.getName() + ", or" +
                     "with the same start date: " + dayOff.getStartDate());
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Comment:
So you agree that you completely change logic.
You are very very wrong about how this method work!
Look again ->
public List<DayOff> findByNameOrStartDate(String name, LocalDate startDate) {
 +        CriteriaBuilder builder = entityManager.getCriteriaBuilder();
 +        CriteriaQuery<DayOff> query = builder.createQuery(DayOff.class);
 +        Root<DayOff> root = query.from(DayOff.class);
 +
 +        query.select(root).where(
 +          builder.or(
 +                  builder.equal(root.get(DayOffConstants.Entity.NAME), name),
 +                  builder.equal(root.get(DayOffConstants.Entity.START_DATE), toDate(startDate))
 +          )
 +        );
 +
 +        return entityManager.createQuery(query).getResultList();
 +    }

And this 
@Override
      public List<DayOff> findByNameOrStartDate(		      public List<DayOff> findByNameOrStartDate(
              String name, LocalDate startDate) {		              String name, LocalDate startDate) {
 -        return dayOffRepository.findByNameOrStartDate(name, startDate);		 +        return dayOffDao.findByNameOrStartDate(name, startDate);
      }		      }
  		  
      @Override		      @Override
      public boolean dayOffExist(String name, LocalDate startDate) {		      public boolean dayOffExist(String name, LocalDate startDate) {
 -        return findByNameOrStartDate(name, startDate).isEmpty();		 +        return !findByNameOrStartDate(name, startDate).isEmpty();
      }		      }